File: Python3/프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    before =words[0][0]
    
    for i, word in enumerate(words):
        if before[-1] != word[0]:
            return [i % n + 1, i // n + 1]
        elif word in words[:i]:
            return [i % n + 1, i // n + 1]
        before = word
    
    return [0, 0]

File: C++/프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <set>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer;
    set<string> wset;
    wset.insert(words[0]);
    
    for(int i = 1; i < words.size(); i++) {
        if(wset.count(words[i]) > 0) {
            return {(i % n) + 1, i / n + 1};
        } else if(words[i].front() != words[i - 1].back()) {
            return {(i % n) + 1, i / n + 1};
        }
        wset.insert(words[i]);
    }
    

    return {0, 0};
}